app/src/main/java/org/p2p/wallet/push_notifications/AppFirebaseMessagingService.kt

```diff
@@ -19,10 +22,13 @@ class AppFirebaseMessagingService : FirebaseMessagingService(), KoinComponent {
     private val appNotificationManager: AppNotificationManager by inject()
     private val appsFlyerService: AppsFlyerService by inject()
     private val intercomPushService: IntercomPushService by inject()
+    private val pushNotificationsInteractor: PushNotificationsInteractor by inject()
+    private val appScope: AppScope by inject()
 
     override fun onNewToken(token: String) {
         appsFlyerService.onNewToken(token)
         intercomPushService.registerForPush(token)
+        appScope.launch { pushNotificationsInteractor.updateDeviceToken() }
```

Comment:
does onNewToken is called on every app launch?

Reply:
no

Reply:
signInByPinCode, signInByBiometric does

Reply:
oh, thanks!